contrib/guix/libexec/build.sh

```diff
@@ -177,14 +177,17 @@ case "$HOST" in
         ;;
 esac
 
+# Unsetting the CMAKE_PREFIX_PATH variable resolves linking issues
+# with bitcoin-qt. This action is possible because other packages
+# using Guix's cmake-build-system continue to build without issues.
```

Comment:
> resolves linking issues # with bitcoin-qt.

Can this be more specific? Why is this only needed in Guix? Is it fixed in later versions of Qt?

> Guix's cmake-build-system continue to build without issues.

Is this guaranteed? Or just happens to work now by chance / might break in the future?

Reply:
> > resolves linking issues # with bitcoin-qt.
> 
> Can this be more specific?

Otherwise, the `build/src/qt/CMakeFiles/bitcoin-qt.dir/link.txt` contains `-L/home/hebasto/.guix-profile/lib` (on my machine), which results in [errors](https://github.com/hebasto/bitcoin/pull/86#pullrequestreview-1847640073) noted by @TheCharlatan.

> Why is this only needed in Guix?

Because Guix sets it to a value which does not work for us. For more details, please refer to https://mail.gnu.org/archive/html/guix-commits/2015-03/msg00518.html.

> Is it fixed in later versions of Qt?

I don't know. Lurking through the https://bugreports.qt.io did not help much.

> > Guix's cmake-build-system continue to build without issues.
> 
> Is this guaranteed? Or just happens to work now by chance / might break in the future?

No guarantees. However, if any package, which uses Guix's cmake-build-system, fails to find its dependencies we still able to specify hint via the [`CMAKE_LIBRARY_PATH`](https://cmake.org/cmake/help/latest/envvar/CMAKE_LIBRARY_PATH.html) and/or [`CMAKE_INCLUDE_PATH`](https://cmake.org/cmake/help/latest/envvar/CMAKE_INCLUDE_PATH.html) environment variables.

---

As an alternative solution, we might consider the patching:
```sh
sed -i "s|-L${CMAKE_PREFIX_PATH}/lib||" build/src/qt/CMakeFiles/bitcoin-qt.dir/link.txt
```

Reply:
> Otherwise, the build/src/qt/CMakeFiles/bitcoin-qt.dir/link.txt contains -L/home/hebasto/.guix-profile/lib (on my machine), 

Ok, so this isn't really guix specific then, other than it's currently only been seen in Guix (i.e I assume a problematic -L could appear outside of a Guix build)? Shouldn't this be worked around by patching Qt in depends?

> Because Guix sets it to a value which does not work for us.

You mean doesn't work for Qt? I assume the exact same problem would happen to anyone (outside of Guix) setting `CMAKE_PREFIX_PATH` to a value Qt doesn't like?

Reply:
Reworked. The `CMAKE_PREFIX_PATH` environment variable is not touched anymore.